pkg/console/operator/sync_v400.go

```diff
@@ -123,10 +124,23 @@ func sync_v400(co *consoleOperator, originalOperatorConfig *operatorv1.Console,
 		}
 	}
 
-	// the operand is available if all resources are present & if we have all the replicas
+	// the operand is available if all resources are:
+	// - present
+	// - if we have all the replicas
+	// - route is admitted
 	// available is currently defined as "met the users intent"
-	if deploymentsub.IsReady(actualDeployment) && routesub.IsAdmitted(rt) {
-		co.ConditionDeploymentAvailable(operatorConfig)
+	if deploymentsub.IsReady(actualDeployment) {
+		if routesub.IsAdmitted(rt) {
+			co.ConditionDeploymentAvailable(operatorConfig)
+		} else {
+			co.SetStatusCondition(
```

Comment:
Here I wasn't sure if we wanna add additional method to the `status.go`, or reuse those that already exist, since we already have quite a lot of them, and don't think that creating a new one for a single usage makes sense.

Reply:
> Here I wasn't sure if we wanna add additional method to the status.go

Yeah, I'd actually prefer removing the existing ones. I think the code is easier to understand with the condition inline like you have here.

Reply:
Sounds like a followUp to me 😄

Reply:
I would like to eliminate them, but also remove all of the inline noise, and instead bubble up a collection of data needed & return from the `sync()` function & create our statuses in one place only.  Long term I don't like these helpers either, but I also don't really want to litter these calls everywhere.